src/loaders/svg/tvgSvgLoader.cpp

```diff
@@ -377,6 +377,27 @@ static char* _idFromUrl(const char* url)
 }
 
 
+static char* _srcFromUrl(const char* url)
+{
+    auto open = strchr(url, '(');
+    auto close = strchr(url, ')');
+    if (!open || !close || open >= close) return nullptr;
+
+    open = strchr(open, '\'');
+    if (!open || open >= close) return nullptr;
+    ++open;
+
+    close = strchr(open, '\'');
+    if (!close || close == open) return nullptr;
+    --close;
+
+    while (open < close && *open == ' ') ++open;
+    while (open < close && *close == ' ') --close;
+
+    return strDuplicate(open, (close - open + 1));
```

Comment:
Hmm... you might wanna consider to avoid data copy...

Reply:
We are in the middle of parsing at this stage. I might not know the font name, for example, so I am unable to load the font yet. so I don't quite see how - could you please elaborate? All char* data that requires it, such as IDs or URLs, is copied.

Reply:
What you need is the pointer and length, rather than a data copy. Based on that, you can decode the base64.

![image](https://github.com/user-attachments/assets/31f87925-0227-4b6e-b302-73172876d783)